src/media.js

```diff
@@ -682,52 +682,55 @@
         },
 
         initPlay() {
-          if (!Object.prototype.hasOwnProperty.call(this, 'Html5AudioRec') && !this.Html5AudioRec) {
-            this.Html5AudioRec = { audioContext: new (window.AudioContext || window.webkitAudioContext)() };
-          }
-
-          if (typeof workletAudioRec !== 'object') {
-            cthis.audio.Html5AudioRec.audioContext.audioWorklet.addModule(workletAudioRecBlob).then(() => {
-              // Setup the connection: Port 1 is for worker 1
-              if (typeof initchannel === 'undefined') {
-                workletAudioRec = new AudioWorkletNode(cthis.audio.Html5AudioRec.audioContext, 'worklet-audio-rec');
-                cthis.audio.Html5AudioRec.MediaStreamDest = cthis.audio.Html5AudioRec.audioContext.createMediaStreamDestination();
-                workletAudioRec.connect(cthis.audio.Html5AudioRec.audioContext.destination);
-
-                virtualclass.gObj.workletAudioRec = workletAudioRec
-                if (virtualclass.system.mybrowser.name === 'Chrome') {
-                  // console.log('==== Chrome after change');
-                  cthis.audio.bug_687574_callLocalPeers();
-                }
-
-                const audioReadyChannel = new MessageChannel();
-                workerIO.postMessage({
-                  cmd: 'workerAudioRec',
-                }, [audioReadyChannel.port1]);
-
-                // Setup the connection: Port 2 is for worker 2
-                workerAudioRec.postMessage({
-                  cmd: 'workerIO',
-                  sampleRate: cthis.audio.Html5Audio.audioContext.sampleRate,
-                }, [audioReadyChannel.port2]);
-
-                const audoPlaychannel = new MessageChannel();
-
-                workerAudioRec.postMessage({
-                  cmd: 'workletAudioRec',
-                }, [audoPlaychannel.port1]);
-
-                // Setup the connection: Port 2 is for worker 2
-                workletAudioRec.port.postMessage({
-                  cmd: 'workerAudioRec',
-                }, [audoPlaychannel.port2]);
-                workerAudioRec.postMessage({ cmd: 'audioWorklet', msg: true });
-                initchannel = true;
-                virtualclass.gObj.audioRecWorkerReady = true;
-              }
-              // virtualclass.gObj.workerAudio = true;
-            });
+          console.log('audio init worklet suman');
+          if (typeof workletAudioRec !== 'undefined') {
+            console.log('audio worklet disconnect');
+            workletAudioRec.disconnect();
           }
+          if (!Object.prototype.hasOwnProperty.call(this, 'Html5Audio') && !this.Html5Audio) {
+            this.Html5Audio = { audioContext: new (window.AudioContext || window.webkitAudioContext)() };
+          }
+
+
+          cthis.audio.Html5Audio.audioContext.audioWorklet.addModule(workletAudioRecBlob).then(() => {
```

Comment:
Codacy found an issue: ['workletAudioRecBlob' is not defined. (no-undef)](https://app.codacy.com/gh/congrea/virtualclass/pullRequest?prid=5978725)